### ISSUE TITLE
Wrong syntax for delete?

### ISSUE THREAD

**Issue by sinnbeck** [inertiajs/inertia#197]
I just noticed that the Inertia.delete method expects (url, data, options). But when actually using it it, is instead  (url, options). The docs even state this as well

```
Inertia.delete(url, {
  replace: false,
  preserveState: false,
  preserveScroll: false,
  only: [],
})
```
Here is how it is shown in phpstorm 
![image](https://user-images.githubusercontent.com/13980140/91713316-d4b68600-eb89-11ea-93d4-aaf661df8719.png)

And index.d.ts
```
delete: (url: string, data?: object, options?: VisitOptions) => Promise<void>

**Comment by claudiodekker** [comment]
Hi,

Very interesting. I wasn't aware of the typescript definition, and assumed (as per the docs & implementation) that the signature was intended to be different.. @reinink?

Continuing on the current implementation as well as assuming that this is intended behaviour, I'd suggest that if you want to pass in data to the method, you can pass it along in to the `options` by doing something like this:
```js
Inertia.delete(url, {
  replace: false,
  preserveState: false,
  preserveScroll: false,
  only: [],
  data: {
    id: user.id
  }
})```

Hope it helps, and thanks again for raising this issue!

**Comment by ycs77** [comment]
The source in TS:

https://github.com/inertiajs/inertia/blob/ac4294e7b47868dc695b1ea23933089bb7bbd60d/packages/inertia/index.d.ts#L56

**Comment by reinink** [comment]
I know, at least in PHP, you cannot send body parameters on a `DELETE` request. That may not be the case for other languages, but that's at least why Inertia's `delete` method only takes a `url` and `options` as its arguments.

Evidently the type definition is wrong? I don't manage the TypeScript file though, this has been managed by @MichaelDeBoey and @sebastiandedeyne.

My plan is to convert Inertia.js to TypeScript fairly soon...which I suspect will take care of these kinds of issues.

**Comment by claudiodekker** [comment]
As per the latest version of the HTTP protocol standard ([RFC 7231](https://tools.ietf.org/html/rfc7231#section-4.3.5)):
>  A payload within a DELETE request message has no defined semantics;
   sending a payload body on a DELETE request might cause some existing
   implementations to reject the request.

Seems to me that based on this the documented signature is correct, and the TypeScript definition to be wrong.

**Comment by reinink** [comment]
This has been fixed (#211). Looks right to me. 👍
